Source: https://src.fedoraproject.org/rpms/mpfi/raw/rawhide/f/mpfi-mismatched-type.patch
--
Fixes these GCC warnings:

init.c:30:19: warning: argument 1 of type '__mpfi_struct[1]' with mismatched bound [-Warray-parameter=]
   30 | mpfi_init (mpfi_t x)
      |            ~~~~~~~^
In file included from mpfi-impl.h:29,
                 from init.c:27:
mpfi.h:62:26: note: previously declared as 'mpfi_ptr' {aka '__mpfi_struct *'}
   62 | void    mpfi_init       (mpfi_ptr);
      |                          ^~~~~~~~

init2.c:30:20: warning: argument 1 of type '__mpfi_struct[1]' with mismatched bound [-Warray-parameter=]
   30 | mpfi_init2 (mpfi_t x, mpfr_prec_t p)
      |             ~~~~~~~^
In file included from mpfi-impl.h:29,
                 from init2.c:27:
mpfi.h:63:26: note: previously declared as 'mpfi_ptr' {aka '__mpfi_struct *'}
   63 | void    mpfi_init2      (mpfi_ptr, mpfr_prec_t);
      |                          ^~~~~~~~

--- a/src/init2.c	2022-03-11 08:33:43.000000000 -0700
+++ b/src/init2.c	2022-03-28 16:23:26.635977528 -0600
@@ -27,7 +27,7 @@ MA 02110-1301, USA. */
 #include "mpfi-impl.h"
 
 void
-mpfi_init2 (mpfi_t x, mpfr_prec_t p)
+mpfi_init2 (mpfi_ptr x, mpfr_prec_t p)
 {
   mpfr_init2 (&(x->left),p);
   mpfr_init2 (&(x->right),p);
--- a/src/init.c	2022-03-11 08:33:43.000000000 -0700
+++ b/src/init.c	2022-03-28 16:21:36.043854934 -0600
@@ -27,7 +27,7 @@ MA 02110-1301, USA. */
 #include "mpfi-impl.h"
 
 void
-mpfi_init (mpfi_t x)
+mpfi_init (mpfi_ptr x)
 {
   mpfr_init (&(x->left));
   mpfr_init (&(x->right));
